### PR TITLE
Respect-`show_avatars`-comments-#2248

### PR DESCRIPTION
**Pull request by BE-Webdesign**
Possible fix for #2248.  Respects the show_avatars option when
retrieving avatars data for comments. Modeled after #2151.

### CODE REVIEW

File: lib/endpoints/class-wp-rest-comments-controller.php [hunk 1]
```diff
@@ -899,6 +885,29 @@ public function get_item_schema() {
 				),
 			),
 		);
+
+		if ( get_option( 'show_avatars' ) ) {
```
Comment (rmccue) [inline lib/endpoints/class-wp-rest-comments-controller.php:892]: This would be better off as a strict check (`=== true`) if we can.

**Comment by rmccue** [comment]
Looks pretty good :+1: Small change in the check, but otherwise seems pretty sensible.

File: lib/endpoints/class-wp-rest-comments-controller.php (hunk 1 above)
Comment (BE-Webdesign) [inline lib/endpoints/class-wp-rest-comments-controller.php:892]: Okay, will fix.  In the codex it is stored as 1 and 0 integer type.  Is this changed to a boolean or should I check for 1 === get_option( 'show_avatars' )?

**Comment by BE-Webdesign** [comment]
Hallelujah!

File: lib/endpoints/class-wp-rest-comments-controller.php [hunk 2]
```diff
@@ -899,6 +885,29 @@ public function get_item_schema() {
 				),
 			),
 		);
+
+		if ( '1' === get_option( 'show_avatars' ) ) {
```
Comment (joehoyle) [inline lib/endpoints/class-wp-rest-comments-controller.php]: I think this should be a `if ( get_option...` the strict type check, while correct for most cases, is somewhat unintuitive, as this _really_ should be a bool. +1 for allowing any truthy value.

**Comment by joehoyle** [comment]
Should we also apply the same to the users controller? The database option name is `show_avatars` which makes it seem like it's not comment specific, though I understand that's where the option is in the dashbaord?

**Comment by BE-Webdesign** [comment]
@joehoyle I believe this was already done in users #2151.  @rmccue suggested the strict check so if == is the way over === we can decide that in the meeting maybe?

**Comment by danielbachhuber** [comment]
I'm :-1: on a strict check in this context. A filter from a plugin or theme should be able to return a truthy or falsy value, and see the API work as expected.

**Comment by BE-Webdesign** [comment]
@danielbachhuber So do you believe we should put it back to:

if ( get_option( 'show_avatars' ) )
**or use**
if (  true == get_option( 'show_avatars' ) )
**or something else entirely?**

**Comment by danielbachhuber** [comment]
I think we should use `if ( get_option( 'show_avatars' ) )`

**Comment by rmccue** [comment]
I'm OK with the truthy/loose check so long as it doesn't cause side effects (i.e. we don't accidentally catch `'n'` or something).

**Comment by danielbachhuber** [comment]
> I'm OK with the truthy/loose check so long as it doesn't cause side effects

Core uses `if ( ! get_option( 'show_avatars' ) )`: https://core.trac.wordpress.org/browser/trunk/src/wp-includes/pluggable.php#L2356

I think we're fine using `if ( get_option( 'show_avatars' ) )`. @BE-Webdesign if you want to make that final change, we can get this landed.

**Comment by BE-Webdesign** [comment]
Okay, I will change it back give me a second.

**Comment by danielbachhuber** [comment]
Build fails because of https://github.com/WP-API/WP-API/issues/1488#issuecomment-184664633

Merging because curies need to be fixed in this release regardless.

**Pull request merged**